### DISCUSSION TITLE
recovering the result of a POST request in a string

### DISCUSSION THREAD

**Discussion by jgalan** [curl/curl#8650] [Q&A] [answered]
I am looking to the following example:

https://github.com/curl/curl/blob/master/docs/examples/http-post.c

I am running it and getting the expected result on screen after executing the following line

```
res = curl_easy_perform(curl);
```

I wonder if there is an easy way to call a similar function that fills in a `std::string` without having to capture `stdout`.

**Comment by jay** [comment, marked as answer]
You'll need to use the write callback function to handle the received data yourself. Check out the [getinmemory](https://curl.se/libcurl/c/getinmemory.html) example.
